### PR TITLE
[FIX/IMP] a couple issues observed during startup

### PR DESCRIPTION
**Pull request by sbidoul**
Fix a couple of issues observed during startup of the job runner on an instance with 500k pending jobs.

### CODE REVIEW

**Comment by sbidoul** [comment]
@sbejaoui @ThomasBinsfeld can you please test this?

**Review by ThomasBinsfeld — APPROVED** [review]
Functional tests :+1:

**Review by sbejaoui — APPROVED** [review]

**Review by guewen — APPROVED** [review]

**Comment by ThomasBinsfeld** [comment]
:rotating_light: Oops. I must have been on the wrong environment. I just detected a bug, sorry! :rotating_light: 

```
2019-03-13 10:31:08,526 8568 INFO ? odoo.addons.queue_job.jobrunner.runner: initializing database connections 
2019-03-13 10:31:08,532 8568 ERROR ? odoo.addons.queue_job.jobrunner.runner: exception: sleeping 5s and retrying 
Traceback (most recent call last):
  File ".../src/odoo12-addon-queue-job/setup/queue_job/odoo/addons/queue_job/jobrunner/runner.py", line 403, in initialize_databases
    for job_data in cr:
TypeError: '_GeneratorContextManager' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../src/odoo12-addon-queue-job/setup/queue_job/odoo/addons/queue_job/jobrunner/runner.py", line 482, in run
    self.initialize_databases()
  File ".../src/odoo12-addon-queue-job/setup/queue_job/odoo/addons/queue_job/jobrunner/runner.py", line 404, in initialize_databases
    self.channel_manager.notify(db_name, *job_data)
  File "/usr/lib/python3.6/contextlib.py", line 185, in __exit__
    self.thing.close()
AttributeError: '_GeneratorContextManager' object has no attribute 'close'
```

**Comment by guewen** [comment]
I merged too fast, didn't see the Work in progress label :/
I created #133 for reverting until a correction.

File: queue_job/jobrunner/runner.py [hunk 1]
```diff
@@ -386,8 +387,9 @@ def initialize_databases(self):
                 _logger.debug('queue_job is not installed for db %s', db_name)
             else:
                 self.db_by_name[db_name] = db
-                for job_data in db.select_jobs('state in %s', (NOT_DONE,)):
-                    self.channel_manager.notify(db_name, *job_data)
+                with closing(db.select_jobs('state in %s', (NOT_DONE,))) as cr:
+                    for job_data in cr:
```
Comment (guewen) [inline queue_job/jobrunner/runner.py:391]: `for job_data in cr.fetchall():`?

**Comment by guewen** [comment]
@sbidoul I'm sorry :disappointed:, I cannot reopen the PR after my erroneous merge. Would you like me to open a new one with the fixup I propose above or will you take care of it?

**Comment by sbidoul** [comment]
No worries, I'll finish this.

Le mer. 13 mars 2019 à 20:45, Guewen Baconnier <notifications@github.com> a
écrit :

**Pull request merged**